firmware/src/ledger/signer/src/main.c

```diff
@@ -136,31 +150,46 @@ void io_seproxyhal_display(const bagl_element_t *element) {
     io_seproxyhal_display_default((bagl_element_t *)element);
 }
 
-// Pick the text elements to display
-static unsigned char display_text_part() {
-    unsigned int i;
-    WIDE char *text = (char *)G_io_apdu_buffer + 5;
-    if (text[current_text_pos] == '\0') {
-        return 0;
-    }
-    i = 0;
-    while ((text[current_text_pos] != 0) && (text[current_text_pos] != '\n') &&
-           (i < MAX_CHARS_PER_LINE)) {
-        lineBuffer[i++] = text[current_text_pos];
-        current_text_pos++;
-    }
-    if (text[current_text_pos] == '\n') {
-        current_text_pos++;
-    }
-    lineBuffer[i] = '\0';
-    return 1;
-}
-
 static void ui_idle(void) {
     uiState = UI_IDLE;
     UX_DISPLAY(bagl_ui_idle_nanos, NULL);
 }
 
+static void ui_screensaver(void) {
+    uiState = UI_SCREENSAVER;
+    UX_DISPLAY(bagl_ui_screensaver_nanos, NULL);
+}
+
+static void handle_button_press(void) {
+    G_idle_time_ms = 0;
+}
+
+static void handle_ticker_event(void) {
+    unsigned int last_idle_time_ms = G_idle_time_ms;
+    G_idle_time_ms += 100;
```

Comment:
That's a good question. This took a bit of experimentation. There is a comment in the SDK that states that the ticker interval is assumed to be 100 ms [here](https://github.com/rsksmart/rsk-powhsm-nanos-secure-sdk/blob/nanos-1314-develop/include/os_io_seproxyhal.h#L423). I did some testing, basically adding a command that returns me the current value of `G_idle_time_ms`, and 100ms is indeed the correct default interval.

We do explicitly set this to `100ms` in the UI app using `io_seproxyhal_setup_ticker`, but there's a comment in this function stating that this shouldn't be used directly by the application since this sets the main ticking source. The comment suggests using `UX_SET_INTERVAL_MS` (which doesn't exist anywhere) instead. This is why I didn't set this explicitly, but on a second look, since this is what we do in the UI, I guess we should probably do the same in the signer to make it consistent, what do you think?

In any case, I agree that this deserves a comment explaining the reasoning, this is not obvious and this info is not easy to find.

Reply:
Ok cool, good to know we know where it comes from. I'm not too fussed about setting it ourselves in the signer, but I would try it and, if it works, leave it there. In any case I would leave a comment explaining the reasoning. I'd also probably factor that `100` into a `#define` just for tidiness' sake.